addons/mail_group/controllers/main.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import werkzeug
+
+from odoo import http, fields, tools
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.addons.portal.controllers.portal import pager as portal_pager
+from odoo.http import request
+from odoo.osv import expression
+
+
+class PortalMailGroup(http.Controller):
+    _thread_per_page = 20
+    _replies_per_page = 5
+
+    def _get_archives(self, group_id):
+        """Return the different date range and message count for the group messages."""
+        Message = request.env['mail.group.message']
+
+        results = Message._read_group_raw(
+            [('mail_group_id', '=', group_id)],
+            ['subject', 'create_date'],
+            groupby=['create_date'], orderby='create_date DESC')
+
+        groups = []
+
+        for result in results:
+            (dates_range, label) = result['create_date']
+            start, end = dates_range.split('/')
+
+            groups.append({
+                'date': label,
+                'date_begin': fields.Date.to_string(fields.Date.to_date(start)),
+                'date_end': fields.Date.to_string(fields.Date.to_date(end)),
+                'messages_count': result['create_date_count'],
+            })
+
+        return groups
+
+    @http.route('/groups', type='http', auth='public', sitemap=True, website=True)
+    def groups(self, email='', **kw):
+        """View of the group lists. Allow the users to subscribe and unsubscribe."""
+        mail_groups = request.env['mail.group'].search([]).sudo()
+
+        if not request.env.user._is_public():
+            # Force the email if the user is logged
+            email_normalized = request.env.user.email_normalized
+            partner_id = request.env.user.partner_id.id
+        else:
+            email_normalized = tools.email_normalize(email)
+            partner_id = None
+
+        members_data = mail_groups._get_members(email_normalized, partner_id)
```

Comment:
It will not work because I first search the group, and then the `_get_members` is done based on the group I have access to, so even if I know the email of someone, I will not be able to know if he's in the group or not (if I do not have the access to the group)

Reply:
The SUDO is done after the search
```python
mail_groups = request.env['mail.group'].search([]).sudo()
```

Reply:
Ah indeed. Have to read code again, but beware to not leak information only using emails on a public route I can spam. I should not be able to get that much level of information so easily.